File: packages/interviewbit/programming/arrays/n3-repeat-number/n3-repeat-number.ts
/**
 * Interviewbit - N3RepeatNumber
 */

function n3RepeatNumber(array: number[]): number {
  const l: number = array.length;
  const count: Map<number, number> = new Map<number, number>();

  array.forEach((n: number) => count.set(n, (count.get(n) || 0) + 1));

  let result: number = -1;
  count.forEach((value: number, key: number) => {
    if (value > Math.floor(l / 3) && result === -1) {
      result = key;
    }
  });

  return result;
}

export { n3RepeatNumber };

File: packages/interviewbit/programming/arrays/n3-repeat-number/n3-repeat-number.py
# pylint: disable=missing-docstring

def n3RepeatNumber(array):
  l = len(array)
  count = {}

  for i in xrange(l):
    count[array[i]] = 1 + count.get(array[i], 0)

  result = -1
  for k, v in count.items():
    print(k, v)
    if v > l / 3:
      result = int(k)

  return result
